lib/chain.js

```diff
@@ -12,63 +12,63 @@ module.exports = (api) => {
   class ArrayChain {
 
     constructor(array) {
-      this._promise = Promise.resolve(array);
+      this.promise = Promise.resolve(array);
```

Comment:
Why expose implementation details? The underscore in the name clearly indicates that it is a private property no one should depend upon as it may change at any time, and taking https://github.com/metarhia/MetaSync/issues/49 into account, such change is likely to happen.